File: VB-Examples/06_Text/GetPageNumberOfText.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Doc
Imports System.IO
Imports Spire.Doc.Documents
Imports Spire.Doc.Pages

Namespace GetPageNumberOfText
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			'Create a new Document object.
			Dim document As New Document()

			'Load a Word document from a file.
			document.LoadFromFile("..\..\..\..\..\..\Data\Sample.docx")

			'Find all occurrences of the word "Spire" in the document and store them in an array of TextSelection objects.
			Dim textSelections() As TextSelection = document.FindAllString("Spire", False, False)

			'Create a new FixedLayoutDocument object using the loaded document.
			Dim layoutDoc As New FixedLayoutDocument(document)

			'Initialize a counter variable.
			Dim count As Integer = 1

			'Create a StringBuilder object to store the result.
			Dim builder As New StringBuilder()

			'Iterate through each TextSelection object in the array.
			For Each selection As TextSelection In textSelections
				'Get the layout entities (lines) associated with the first range in the TextSelection.
				For Each line As FixedLayoutSpan In layoutDoc.GetLayoutEntitiesOfNode(selection.GetRanges()(0))
					'Get the page index of the line.
					Dim index As Integer = line.PageIndex

				'Append the result to the StringBuilder.
				builder.AppendLine("The matched word " & count & " is on page:" & index)
				
				'Increment the counter.
				count += 1
			Next line
			Next selection

			'Save the result to a text file.
			File.WriteAllText("result.txt", builder.ToString())

			'Dispose of the Document object.
			document.Dispose()
			
			Process.Start("result.txt")
		End Sub

	End Class
End Namespace

File: CS-Examples/06_Text/GetPageNumberOfText.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Doc;
using System.IO;
using Spire.Doc.Documents;
using Spire.Doc.Pages;

namespace GetPageNumberOfText
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {   
			// Create a new Document object
			Document document = new Document();

			// Load a Word document from the specified file path
			document.LoadFromFile(@"..\..\..\..\..\..\Data\Sample.docx");

			// Find all occurrences of the string "Spire" in the document
			TextSelection[] textSelections = document.FindAllString("Spire", false, false);

			// Create a FixedLayoutDocument object using the loaded document
			FixedLayoutDocument layoutDoc = new FixedLayoutDocument(document);

			// Initialize a counter for matched words
			int count = 1;

			// Create a StringBuilder to store the result
			StringBuilder builder = new StringBuilder();

			// Iterate through each TextSelection
			foreach (TextSelection selection in textSelections)
			{
				// Get the layout entities for the current selection
				foreach (FixedLayoutSpan line in layoutDoc.GetLayoutEntitiesOfNode(selection.GetRanges()[0]))
				{
					// Get the page index where the matched word is located
					int index = line.PageIndex;

					// Append the result to the StringBuilder
					builder.AppendLine("The matched word " + count + " is on page:" + index);

					// Increment the counter
					count++;
				}
			}

			// Write the result to a text file named "result.txt"
			File.WriteAllText("result.txt", builder.ToString());

			// Dispose the Document object to release resources
			document.Dispose();
			
            System.Diagnostics.Process.Start("result.txt");
        }

    }
}
